evap/results/views.py

```diff
@@ -63,13 +63,17 @@ def course_detail(request, semester_id, course_id):
     sections = calculate_results(course)
 
     if request.user.is_reviewer:
-        public_view = request.GET.get('public_view') != 'false'  # if parameter is not given, show public view.
+        view = request.GET.get('view', 'public')  # if parameter is not given, show public view.
     else:
-        public_view = request.GET.get('public_view') == 'true'  # if parameter is not given, show own view.
+        view = request.GET.get('view', 'full')  # if parameter is not given, show own view.
+
+    contributor_id = request.user.id
+    if view == 'filter' and user.is_staff:
+        contributor_id = int(request.GET.get('contributor_id'))
 
     # If grades are not published, there is no public view
-    if not course.can_publish_grades:
-        public_view = False
+    if not course.can_publish_grades and view == 'public':
+        view = 'full'
 
     represented_users = list(request.user.represented_users.all())
     represented_users.append(request.user)
```

Comment:
`request.user` is used in various places where i think it would be correct to use the user passed via the contributor_id parameter.